src/web/js/output-ui.js

```diff
@@ -612,70 +394,46 @@
       });
     }
     
-    function getSourceContent(editors, cmloc, tight) {
-      if(editors.hasOwnProperty(cmloc.source)) {
-        if(!tight) {
-          cmloc =
-            {start:{
-              line: cmloc.start.line,
-              ch: 0},
-             end:{
-              line: cmloc.end.line,
-              ch: editors[cmloc.source].getLine(cmloc.end.line).length}};
-        }
-        return editors[cmloc.source].getRange(cmloc.start, cmloc.end);
-      } else {
-        var source = sessionStorage.getItem(cmloc.source);
-        if(!source) {
-          return undefined;
-        }
-        var lines = source.split("\n").slice(cmloc.start.line, cmloc.end.line + 1);
-        if(tight) {
-          lines[lines.length - 1] = lines[lines.length - 1].substr(0, cmloc.end.ch);
-          lines[0] = lines[0].substr(cmloc.start.ch);
-        }
-        return lines.join("\n");
-      }
-    }
-    
-    function makeMaybeLocToAST(runtime, editors, srcloc) {
+    function makeMaybeLocToAST(runtime, documents, srcloc) {
       return runtime.makeFunction(function(loc) {
         return runtime.ffi.cases(runtime.getField(srcloc, "is-Srcloc"), "Srcloc", loc, {
           "builtin": function(_) {
             console.error("maybeLocToAST should not be passed a builtin source location.", loc);
             return runtime.ffi.makeNone();
           },
-          "srcloc": function(filename, start_line, start_col, startCh, endL, endC, endCh) {
+          "srcloc": function(filename, start_line, start_col, _, end_line, end_col, _) {
```

Comment:
It's allowed and well addressed in the specification, though this particular use is disallowed on strict-mode functions. I removed the names after a particularly frustrating bug caused by a naming inconsistency: CodeMirror positions refer to the the text column component of the position as `ch`, but in Pyret `srcloc` values, the `ch` fields are indices into the characters of the entire document.

(I think I recall the `ch` fields of pyret srclocs actually being incorrect, but I can't recall how I encountered that... Regardless, they're of absolutely no use.)

I could rename the underscores `_start_ch` and `_end_ch` following the prolog convention of `_name` indicating an unused and ignored parameter, but in javascript the convention  of prefixing names with an underscore usually means something entirely different.